tasks.py

```diff
@@ -184,6 +168,63 @@ def deploy(c, hosts=""):
     deploy_nixos(get_hosts(hosts))
 
 
+def decrypt_host_keys(c, hostname, tmpdir):
+    os.mkdir(f"{tmpdir}/etc")
+    os.mkdir(f"{tmpdir}/etc/ssh")
+    os.umask(0o177)
+    c.run(
+        f"sops --extract '[\"ssh_host_ed25519_key\"]' --decrypt {ROOT}/{hostname}/secrets.yaml > {tmpdir}/etc/ssh/ssh_host_ed25519_key"
+    )
+
+
+@task
+def print_keys(c, hosts=""):
+    """
+    Decrypt host private key, print ssh and age public keys. Use inv print-host-keys --hosts build01
+    """
+    g = DeployGroup(get_hosts(hosts))
+
+    def key(h: DeployHost) -> None:
+        hostname = h.host.replace(".nix-community.org", "")
+        with tempfile.TemporaryDirectory() as tmpdir:
+            decrypt_host_keys(c, hostname, tmpdir)
+            pubkey = subprocess.run(
+                ["ssh-keygen", "-y", "-f", f"{tmpdir}/etc/ssh/ssh_host_ed25519_key"],
+                stdout=subprocess.PIPE,
+                text=True,
+                check=True,
+            )
+            print("###### Public keys ######")
+            print(pubkey.stdout)
+            print("###### Age keys ######")
+            subprocess.run(
+                ["ssh-to-age"],
+                input=pubkey.stdout,
+                check=True,
+                text=True,
+            )
+
+    g.run_function(key)
+
+
+@task
+def install(c, hosts=""):
+    """
+    Decrypt host private key, install with nixos-anywhere. Use inv install --hosts build01
+    """
+    g = DeployGroup(get_hosts(hosts))
+
+    def anywhere(h: DeployHost) -> None:
+        hostname = h.host.replace(".nix-community.org", "")
+        with tempfile.TemporaryDirectory() as tmpdir:
+            decrypt_host_keys(c, hostname, tmpdir)
+            c.run(
+                f"nix run github:numtide/nixos-anywhere#nixos-anywhere -- --extra-files {tmpdir} --flake .#{hostname} {h.host}"
```

Comment:
This won't work with aarch64-linux as the default kexec tarball is x86_64.

@Mic92 Could nixos-anywhere be changed to read the system from the flake and set the right tarball?

Reply:
Sounds like a good idea until $someone finalizes aarch64 ci.